src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -1027,4 +1027,12 @@ public String getApiCallableTypeName(ServiceMethodType serviceMethodType) {
   public String getReleaseAnnotation(ReleaseLevel releaseLevel) {
     return getNotImplementedString("SurfaceNamer.getReleaseAnnotation");
   }
+
+  public String getMetadataIdentifier() {
+    return getNotImplementedString("SurfaceNamer.getMetadataIdentifier");
+  }
+
+  public String getMetadataName() {
```

Comment:
Done. Added NodeJSPackageMetadataNamer. I found a parent `PackageMetadataNamer` is not necessary so far since it is only used in the language specific transformer.